gsa_components/gt_u7/gt_u7.py

```diff
@@ -2,74 +2,67 @@
 import pynmea2
 from threading import Thread
 import gps
+import os
+from time import sleep
 
 class Gt_u7:
-    def __init__(self, serial_port="/dev/ttyACM0", baudrate=9600):
-        self.ser = serial.Serial(serial_port, baudrate)
+    def __init__(self, serial_port:str="/dev/ttyACM0", baudrate:int=9600)->None:
+        self.serial_port = serial_port
+        self.baudrate = baudrate
         self.latitude = None
         self.longitude = None
         self.altitude = None
-        self.stop_thread = False
-        self.thread = Thread(target=self.read_data_thread)
+        self.thread = Thread(target=self.start)
         self.thread.start()
 
-    def read_data(self):
-        try:
-            while not self.stop_thread:
+    def start(self)->None:
+        self.stop_thread = False
+        self.ser = serial.Serial(self.serial_port, self.baudrate)
+        self.read_data()
+
+    def read_data(self)->None:
+        while not self.stop_thread:
+            try:
                 data = self.ser.readline().decode().strip()
                 if data.startswith("$GPGGA"):
                     msg = pynmea2.parse(data)
```

Comment:
i mean ich checke das wir sowieso das gleiche machn müssen ob jetzt nur pynmea2 failed oder schon data = slef.ser... aber es wäre eig cleaner wenn das auch einen try hätte oder? Weil das ist ja was, dass errors thrown kann und es wäre cleaner von einem debug point of view wenn es seinen eigene try hat. Aber idk, so funktioniert es sicher auch

Reply:
es wird aus dem error code ersichtlich ob es bei data war oder pynmea2, bin alle errors durchgegangen. (beim ersten steht so: Serial reading failed: blablabla, und beim zweiten: pynmea2-error: blablabal). darum würd ichs so lassen